pkg/config/install_requirements.go

```diff
@@ -243,6 +243,22 @@ type IngressConfig struct {
 	DomainIssuerURL string `json:"domainIssuerURL,omitempty"`
 }
 
+// BuildPackConfig contains build pack info
+type BuildPackConfig struct {
+	// Location contains location config
+	BuildPackLibrary BuildPackLibrary `json:"buildPackLibrary,omitempty"`
```

Comment:
Aaaaah - re my comment about the staging pipeline using the version stream image and failing with 
```
Showing logs for build cb-kubecd/environment-pr-7172-113-boot-vault-staging/master #1 promotion stage from-build-pack and container step-setup-jx-git-credentials
error: error getting Requirements from TeamSettings to determine if in GitHub app mode: validation failures in requirements from team settings:
(root): Additional property buildPacks is not allowed
```

The requirements has
```
      buildPacks:
        buildPackLibrary: {}
```
in it, because `BuildPackLibrary` is never "empty" per se, since it's not a pointer. So I think if this is changed to a pointer, it won't be put in the serialized requirements file if there's nothing in it.

Reply:
^^^ @hferentschik - does that sound right?

Reply:
jupp, good catch.

Reply:
> . So I think if this is changed to a pointer, it won't be put in the serialized requirements file if there's nothing in it.

I think it is actually `BuildPackConfig` which needs to be a pointer. See also `AzureConfig` and `GKEConfig`. Easy to test. Create an empty `RequirementsConfig`, serialize it, deserialize it and ensure that the ``BuildPackConfig` is nil. And for good measure one can assert against the generated file as well.

Reply:
@mrageh - does that make sense? Please add the test as well, to make sure we don't end up with another surprise.

Reply:
@mrageh - seems tests are passing now. Are you still working on this additional test or do you want to merge w/o?